app/src/main/java/com/example/habithelper/activities/SignupActivity.java

```diff
@@ -0,0 +1,112 @@
+package com.example.habithelper.activities;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.example.habithelper.R;
+import com.parse.LogInCallback;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+public class SignupActivity extends AppCompatActivity {
+
+    private EditText etUsername;
+    private EditText etPassword;
+    private Button btnCreate;
+    private String username;
+    private String password;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_signup);
+
+        etUsername = findViewById(R.id.etUsername);
+        etPassword = findViewById(R.id.etPassword);
+        btnCreate = findViewById(R.id.btnCreate);
+
+        btnCreate.setOnClickListener(new View.OnClickListener() {
+
+            /**
+             * Attempts to sign up a ParseUser with the username and password user input
+             * @param v the view that was clicked
+             */
+            @Override
+            public void onClick(View v) {
+                username = etUsername.getText().toString();
+                password = etPassword.getText().toString();
+
+                checkValid();
+                login();
+            }
+        });
+    }
+
+    /**
+     * Attempts to login the newly-created ParseUser and navigate to the AccountCreationActivity
+     */
+    private void login() {
+        ParseUser.logInInBackground(username, password, new LogInCallback() {
+            @Override
+            public void done(ParseUser user, ParseException e) {
+                if (e != null) {
+                    Toast.makeText(SignupActivity.this, "Issue with login!", Toast.LENGTH_SHORT);
+                    return;
+                }
+                Intent i = new Intent(SignupActivity.this, AccountCreationActivity.class);
```

Comment:
Since the variable i is not used anywhere else, you can also do:
`startActivity(new Intent(SignupActivity.this, AccountCreationActivity.class));`